orchagent/intfsorch.cpp

```diff
@@ -1219,11 +1220,40 @@ void IntfsOrch::removeRifFromFlexCounter(const string &id, const string &name)
     SWSS_LOG_DEBUG("Unregistered interface %s from Flex counter", name.c_str());
 }
 
+void IntfsOrch::cleanUpRifFromCounterDb(const string &id, const string &name)
```

Comment:
Could you consider moving this to syncd?

Reply:
Thanks @Junchao-Mellanox for the review/comments. The current changes done is alongside/inline-with other rif counter  cleanup code. The idea was to have all rif counter related cleanup should be triggered at the same place. Please refer to the call made for removeRifFromFlexCounter in removeRouterIntfs .
Will also request @dgsudharsan for comments if all the clean-up for rif counters (existing:removeRifFromFlexCounter and newly added cleanUpRifFromCounterDb) should move to syncd ?

Reply:
@Junchao-Mellanox , This request (to move cleanup to syncd) needs to be evaluated and then considered . Please note that there are other cleanup for rif counters which also needs to be moved to syncd in such case. 
For now , will move ahead with this PR merge.  If after discussion/evaluation it is decided to move these code to syncd , will have a new PR for the same.

Reply:
@sumanbrcm I believe the reason for requesting to move is due to a race condition when orchagent deletes the counters, syncd can try to access it given the operations are asynchronous. I suggest to have the cleanup of counters in syncd but if this warrants a discussion, lets have one. @prsunny What is your opinion on this?

Reply:
Since this race condition won't affect any function, please add a TODO comment here to remind there is a race condition. I will approve the PR after that.

Reply:
@sumanbrcm , @Junchao-Mellanox, since the tests are passing, I'm going ahead with the merge as the outstanding concern is to add comment. @sumanbrcm , please raise another PR to address the comment.